File: rightSideViewDFS.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.result = []
    
    def rightSideView(self, root: TreeNode) -> List[int]:
        
        if root is None:
            return None
        
        self.dfs(root, 0)
        
        return self.result
        
        
    def dfs(self, root: TreeNode, level: int):
        
         #base
            if root is None:
                return None
            
            #logic
            if level == len(self.result):
                self.result.append(root.val)

            
            self.dfs(root.right, level+1)
            self.dfs(root.left, level+1)
            
#  DFS solution.
            

            
        
            
            
    
        
        

File: rightsideviewdfs.java
//time : O(N)
//Space : O(N)

class Solution {
    List<Integer> result;
    public List<Integer> rightSideView(TreeNode root) {
        result = new ArrayList<>();
        if(root == null)
            return result;
        dfs(root,0);
        return result;
    }
    
    private void dfs(TreeNode root , int level)
    {
        if(root == null)
            return;
            if(result.size()==level)
                result.add(root.val);
                else
                result.set(level,root.val);
        dfs(root.left,level+1);
        dfs(root.right,level+1);
    }
}